enacts/calc.py

```diff
@@ -1,28 +1,126 @@
 import numpy as np
 import pandas as pd
 import xarray as xr
+import datetime
 
 # Date Reading functions
-def read_zarr_data(zarr_path):
-    """Read and return data in zarr format.
+
+np.random.seed(123)
+
+def get_data(variable, time_res, ds_conf):
+    """ Gets ENACTS data for ENACTS Maprooms, read from files or synthetic
+
+     Parameters
+    ----------
+    variable : str
+        string representing ENACTS variable ("precip", "tmin" or "tmax")
+    time_res : str
+        "daily" or "dekadal" resolution of the desired variable
+    ds_conf : dict
+        dictionary indicating ENACTS datasets configuration
+        (see config)
+    
+    Returns
+    -------
+        `xr.DataArray` of ENACTS `variable` for `time_res` time step
+    
+    See Also
+    --------
+    read_enacts, synthetize_enacts
+    """
+    if ds_conf[time_res] == "FAKE" :
+        return synthetize_enacts(variable, time_res)
+    else:
+        return read_enacts(variable, ds_conf[time_res])
+
+
+def read_enacts(variable, dst_conf):
+    """ Read ENACTS data
 
     Parameters
-    ------
-    zarr_path : str
-        String of path to zarr folder.
+    ----------
+    variable : str
+        string representing ENACTS variable ("precip", "tmin" or "tmax")
+    dst_conf : dict
+        dictionary indicating ENACTS zarr paths for a given time resolution
+        (see config)
+    
     Returns
     -------
-    zarr_data : Dataset
-        Data from zarr folder as multidimensional xarray dataset.
+        `xr.DataArray` of ENACTS `variable` for `dst_conf` time step
+    
     See Also
     --------
-    Notes
-     -----
+    xr.open_zarr
     """
-    zarr_data = xr.open_zarr(zarr_path)
-    return zarr_data
+    data_path = dst_conf['vars'][variable][1]
+    if data_path is None:
+        data_path = dst_conf['vars'][variable][0]
+    zarr_path = f"{dst_conf['zarr_path']}{data_path}"
+    var_name = dst_conf['vars'][variable][2]
+    xrds = xr.open_zarr(zarr_path)
+    return xrds[var_name]    
 
 
+def synthetize_enacts(variable, time_res):
```

Comment:
It's spelled (and pronounced) "synthesize," with an s. ("Synthetic" with a t is correct, however.)